### PR TITLE
Add unit tests

### LINKED ISSUE #5: Add unit testing and continuous integration
There are currently no tests in the ICA-AROMA code. We can add unit tests and continuous integration with CircleCI to ensure that we don't introduce any breaking changes. The good news is that, since we're working off of functional code, we can run ICA-AROMA on a test dataset to generate outputs, and then compare against those as our "ground truth."

### PR DESCRIPTION
**Pull request by eurunuela**
Closes #5

- Update `setup.cfg` file to make CircleCI work.
- Add unit tests for functions that do NOT depend on FSL.
- Make integration test compare results with results obtained with the original workflow.

### CODE REVIEW

**Comment by tsalo** [comment]
For the integration test, we could use a Docker image with FSL installed, I guess? I am not sure if we want to go to the effort of getting that to pass when our goal is to drop FSL entirely.

**Comment by eurunuela** [comment]
I'd run the integration test locally until we completely drop FSL. I don't think the docker is worth the effort.

**Comment by tsalo** [comment]
Local checks sound good to me.

**Comment by eurunuela** [comment]
Right now we cannot write any other unit tests without using FSL.

Shall we merge this and add new unit tests as we replace FSL with Python code?

**Comment by tsalo** [comment]
Can you add the other outputs from MELODIC so we can test the feature functions? The `melodic_FTmix`, `melodic_mix`, and 4D thresholded component image (which I think AROMA creates from the individual 3D thresholded component images) should be sufficient.

**Comment by eurunuela** [comment]
> Can you add the other outputs from MELODIC so we can test the feature functions? The `melodic_FTmix`, `melodic_mix`, and 4D thresholded component image (which I think AROMA creates from the individual 3D thresholded component images) should be sufficient.

I think I'm only missing the 4D thresholded component image. I'll add it now.

File: aroma/tests/test_features.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    true_maxRPcorr = np.array([0.65255575, 0.86003032, 0.88690363, 0.61399576, 0.43840624])
+
+    assert np.allclose(maxRPcorr[:len(true_maxRPcorr)], true_maxRPcorr, atol=1e-2)
+
+
+def test_feature_frequency(nilearn_data):
+
+    # Get path to melmix file
+    cwd = utils.get_resource_path()
+    melTmix = op.join(cwd, 'aroma', 'resources', 'melodic_FTmix')
```
Comment (tsalo) [inline aroma/tests/test_features.py:39]: No need for the second set of `"aroma", "resources"`.

Comment (eurunuela) [reply aroma/tests/test_features.py:39]: True! Thanks!

**Comment by tsalo** [comment]
Ultimately, I'd like to move the test data into a subfolder of `tests/`, with its own `tests`-specific `utils.py` containing a `get_test_data_path()` function, but that doesn't have to go in this PR if you don't want to do it now.

**Comment by eurunuela** [comment]
> Ultimately, I'd like to move the test data into a subfolder of `tests/`, with its own `tests`-specific `utils.py` containing a `get_test_data_path()` function, but that doesn't have to go in this PR if you don't want to do it now.

What would you like to store there? The original workflow results?

**Comment by tsalo** [comment]
> What would you like to store there? The original workflow results?

Exactly. Plus any other data we plan to test against, but which isn't useful for the package as a whole. At the moment that's just the original workflow results, but we could add more stuff there over time.

**Comment by eurunuela** [comment]
Will add it to this PR before I switch off for the day.

**Review by tsalo — APPROVED** [review]
Looks great to me!

**Pull request merged**